File: 0784-insert-into-a-binary-search-tree/0784-insert-into-a-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode insertIntoBST(TreeNode root, int val) {
        if(root == null){
            return new TreeNode(val);
        }

        TreeNode temp = root;
        while(true){
            if(temp.val < val){
                if(temp.right == null){
                    temp.right = new TreeNode(val);
                    break;
                } else {
                    temp = temp.right;
                }
            } else {
                if(temp.left == null){
                    temp.left = new TreeNode(val);
                    break;
                } else {
                    temp = temp.left;
                }
            }
        }

        return root;
    }
}

File: 0784-insert-into-a-binary-search-tree/0784-insert-into-a-binary-search-tree.py
class Solution:
    def insertIntoBST(self, root, val):
        if not root:
            return TreeNode(val)

        temp = root
        while True:
            if val > temp.val:
                if not temp.right:
                    temp.right = TreeNode(val)
                    break
                temp = temp.right
            else:
                if not temp.left:
                    temp.left = TreeNode(val)
                    break
                temp = temp.left

        return root
